src/v/coproc/router.cc

```diff
@@ -24,8 +24,12 @@
 #include <seastar/core/gate.hh>
```

Comment:
You write "during [...] run loop, internal state isn't being mutated" but the subject is "defer mutations [..] to occur _within_ run loop", which I think needs some clarification. If it just mutations, then why write "causing potential lifetime errors" but if it is lifetime issues, then shouldn't the the subject indicate that operations are deferred until _after_ the run loop?

Reply:
that comment was removed in a later commit in the patch

Reply:
it wasn't a comment, it was a commit message. but it looks like you updated it. thanks!

Reply:
ah great!